src/s2geography/projections.cc

```diff
@@ -32,11 +36,19 @@ class OrthographicProjection: public S2::Projection {
   R2Point Project(const S2Point& p) const {
     S2Point out = S2::Rotate(p, z_axis_, -centre_.lng());
     out = S2::Rotate(out, y_axis_, centre_.lat());
-    return R2Point(out.y(), out.z());
+    if (out.x() >= 0) {
+      return R2Point(out.y(), out.z());
+    } else {
+      return R2Point(NAN, NAN);
+    }
   }
 
   // Converts a projected 2D point to a point on the sphere.
   S2Point Unproject(const R2Point& p) const {
+    if (std::isnan(p.x()) || std::isnan(p.y())) {
+      throw Exception("Can't unproject orthographic for non-finite point");
```

Comment:
This error message is a bit confusing if you get this from a s2geography -> projected geoarrow conversion, because you are projecting in that case, not unprojecting. The fact that under the hood it unprojects for `EstimateMaxError` is not something the user can know.

Now, not sure if there is an easy way to avoid that (because it could also be called just for Unproject with invalid nan values), and for spherely we can catch the error and provide a better one

Reply:
> In spherely if you're doing an orthographic projection you also will want to run it by an intersection of the hemisphere (which will eliminate points that would return this value):
> 
> https://github.com/r-spatial/s2/blob/main/R/plot.R#L77-L96

Yep, was just working on that :)